sbncode/TPCReco/VertexStub/PlaneTransform.h

```diff
@@ -0,0 +1,89 @@
+#ifndef PlaneTransform_HH
+#define PlaneTransform_HH
+
+#include "art/Framework/Principal/Event.h"
+#include "art/Framework/Principal/Handle.h"
+#include "canvas/Persistency/Common/FindManyP.h"
+
+#include "lardataobj/RecoBase/PFParticle.h"
+#include "lardataobj/RecoBase/Cluster.h"
+#include "lardataobj/RecoBase/Slice.h"
+#include "lardataobj/RecoBase/Vertex.h"
+#include "lardataobj/RecoBase/Track.h"
+#include "lardataobj/RecoBase/TrackTrajectory.h"
+#include "lardataobj/RecoBase/Trajectory.h"
+#include "lardataobj/RecoBase/Wire.h"
+#include "lardataobj/RecoBase/Hit.h"
+#include "lardataobj/RecoBase/TrackHitMeta.h"
+#include "larcore/Geometry/Geometry.h"
+#include "larcorealg/Geometry/GeometryCore.h"
+#include "lardataalg/DetectorInfo/DetectorPropertiesStandard.h"
+#include "lardata/DetectorInfoServices/DetectorPropertiesService.h"
+#include "larreco/Calorimetry/CalorimetryAlg.h"
+#include "sbnobj/Common/Reco/VertexHit.h"
+#include "sbnobj/Common/Reco/Stub.h"
+
+#include <memory>
+
+namespace sbn {
+
+// Code for transforming plane coordinates into 3D coordinates
+//
+// Copied liberally from the Pandora LArRotationalTransformationPlugin
+class PlaneTransform {
+public:
+  PlaneTransform(fhicl::ParameterSet const& p);
+  double UVtoW(const double u, const double v) const;
+  double VWtoU(const double v, const double w) const;
+  double WUtoV(const double w, const double u) const;
+  
+  double UVtoY(const double u, const double v) const;
+  double UVtoZ(const double u, const double v) const;
+  double UWtoY(const double u, const double w) const;
+  double UWtoZ(const double u, const double w) const;
+  double VWtoY(const double v, const double w) const;
+  double VWtoZ(const double v, const double w) const;
+  
+  double YZtoU(const double y, const double z) const;
+  double YZtoV(const double y, const double z) const;
+  double YZtoW(const double y, const double z) const;
```

Comment:
@PetrilloAtWork may know better, but I think there should be Geometry utilities that can do these, rather than making new functions and relying on a properly configured fcl config. Is it possible to rely on the geometry service here?

Reply:
I looked and could not find any Geometry functions that provided this utility. Do you know where it would be?

Reply:
Yeah, @PetrilloAtWork should definitely weigh in ... but I think at least grabbing wire angle in a plane is here:

https://github.com/SBNSoftware/larcorealg/blob/develop/larcorealg/Geometry/PlaneGeo.h#L513

and PlaneGeo has a lot of other utilities to use I think ... I'm not sure this is exactly these functions, but there's a lot to be able to pull from the geometry itself rather than rely on fcl config.

Even while leaving these functions/etc., I think at a minimum it'd be good to not have specified in fcl the different angles and views, as that should be coming from the geometry itself.

Reply:
I also have a little trouble wrapping my head around this stuff, but I don't believe that this code makes that assumption. 

This is because the "Wire-Coordinate" that is passed to these functions is not the same as the wire number. The wire coordinate of a wire is obtained from the sbn::PlaneTransform::WireCoordinate function, which explicitly pulls from the wire geometry. This means that the origin used in the plane matching will be the same as the origin in the detector geometry -- it is not assumed to be in any special location.